front-end/src/context/tokenContext.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, useContext, useState, useEffect} from "react"
+import {baseURL, client} from "../config/AxiosConfig";
+import Cookies from 'universal-cookie';
+
+export type userTokenContent = {
+  user_token: string
+  setUserToken:(c: string) => void
+}
+
+export const MyTokenContext = createContext<userTokenContent>({
+    user_token: '',
+    setUserToken: () => {},
+})
+
+export default function Provider({children}:any) {
+    const cookies = new Cookies();
+    const [user_token, setUserToken] = useState<string>(cookies.get('user_token'))
+    const headers = {'Authorization': 'Bearer '+user_token}
+
+    const isTokenExpired = async (user_token:string) => {
+        const cookies = new Cookies();
+        const response: any = await client.get(baseURL+`/users/token_expired`,{headers: headers});
```

Comment:
the response variable in the isTokenExpired function is not very descriptive - it might be better to call it something like tokenCheckResponse to indicate that it's related to checking the validity of the user token.